docs/core/metrics.md

```diff
@@ -0,0 +1,176 @@
+
+!!! note "Metrics overflow"
+    CloudWatch EMF supports a max of 100 metrics. Metrics utility will flush all metrics when adding the 100th metric while subsequent metrics will be aggregated into a new EMF object, for your convenience.
+
+### Flushing metrics
+
+The `@Metrics` annotation **validates**, **serializes**, and **flushes** all your metrics. During metrics validation, 
+if no metrics are provided no exception will be raised. If metrics are provided, and any of the following criteria are 
+not met, `ValidationException` exception will be raised.
+
+!!! tip "Metric validation"
+    * Minimum of 1 dimension
+    * Maximum of 9 dimensions
+
+If you want to ensure that at least one metric is emitted, you can pass `raiseOnEmptyMetrics = true` to the **@Metrics** annotation:
+
+=== "App.java"
+
+    ```java hl_lines="1"
+    @Metrics(raiseOnEmptyMetrics = true)
+    public Object handleRequest(Object input, Context context) {
+    ...
+    }
+    ```
+
+## Capturing cold start metric
+
+You can capture cold start metrics automatically with `@Metrics` via the `captureColdStart` variable.
+
+=== "App.java"
+
+    ```java hl_lines="1"
+    @Metrics(captureColdStart = true)
+    public Object handleRequest(Object input, Context context) {
+    ...
+    }
+    ```
+
+If it's a cold start invocation, this feature will:
+
+* Create a separate EMF blob solely containing a metric named `ColdStart`
+* Add `FunctionName` and `Service` dimensions
+
+This has the advantage of keeping cold start metric separate from your application metrics.
+
+## Advanced
+
+## Adding metadata
+
+You can use `putMetadata` for advanced use cases, where you want to metadata as part of the serialized metrics object.
+
+!!! info
+    **This will not be available during metrics visualization, use `dimensions` for this purpose.**
+
+=== "App.java"
+
+    ```java hl_lines="3 4" 
+    @Metrics(namespace = "ServerlessAirline", service = "payment")
+    public APIGatewayProxyResponseEvent handleRequest(Object input, Context context) {
+        metricsLogger().putMetric("CustomMetric1", 1, Unit.COUNT);
+        metricsLogger().putMetadata("booking_id", "1234567890");
+        ...
+    }
+    ```
+
+This will be available in CloudWatch Logs to ease operations on high cardinal data.
+
+## Creating a metric with a different dimension
+
+CloudWatch EMF uses the same dimensions across all your metrics. Use `withSingleMetric` if you have a metric that should have different dimensions.
+
+!!! info
+    Generally, this would be an edge case since you [pay for unique metric](https://aws.amazon.com/cloudwatch/pricing/). Keep the following formula in mind:
+    **unique metric = (metric_name + dimension_name + dimension_value)**
+
+=== "App.java"
+
+    ```java
+    withSingleMetric("CustomMetrics2", 1, Unit.COUNT, "Another", (metric) -> {
+        metric.setDimensions(DimensionSet.of("AnotherService", "CustomService"));
+    });
+    ```
```

Comment:
This one is not like the others 😄 . I assume it is within the handler method, but it is not clear where this snipped should be in the code. Having it in the scope of the handler would help to understand where it can/should be used.